pybind11/how-to.rst

```diff
@@ -166,82 +166,72 @@ Adding dependencies
 First we need to add some dependencies to the build.
 
 Scons will not use pybind11 unless it is setup, so in ``{{pkg}}/ups/{{pkg}}.table``,
-where ``{{pkg}}`` is the name of the package, you will need to add the dependency
-``setupRequired(pybind11)``.
+where ``{{pkg}}`` is the name of the package, you will need to add the dependency ``setupRequired(pybind11)``.
 You also need to modify the ``dependencies`` in ``{{pkg}}/ups/{{pkg}}.cfg``, by adding ``"pybind11"`` to ``"buildRequired"``.
 
 .. _creating-module-file:
 
 Creating a module file
 ^^^^^^^^^^^^^^^^^^^^^^
 
-Following :ref:`our rules on file naming <style-guide-pybind11-module-naming>`, we start by creating a minimal module file ``python/lsst/TMPL/exampleOne.cc`` with the following content:
+Following :ref:`our rules on file naming <style-guide-pybind11-module-naming>`, we start by creating a minimal module file ``python/lsst/TMPL/_tmpl.cc`` with the following content:
```

Comment:
Wouldn't it be better to be consistent about one-cc-file-per-include, instead _adding_ a `_tmpl.cc` that calls the `wrapExampleOne` function?

Reply:
I'm not quite sure what your alternate proposal is.  When there are multiple per-header .cc files I don't think it's appropriate to just pick one to drop `PYBIND11_MODULE` in.  But I also don't want to mandate that tiny packages with just one header (e.g. `meas_extension_photometryKron`) have an extra package-level .cc file with just the `PYBIND11_MODULE`.  The problem with this example, as I see it, is that it starts out being a tutorial for the latter case (to start out simple) and then turns into a tutorial for the former case.

Reply:
I was indeed suggesting that every package have a `_package.cc` file, even if they had only one include file, to make things consistent.  It depends on what you see as the usual case.  If most packages have multiple includes, then I'd say start by describing that case and treat the only-one-include as a special optimization.  If you think most packages will have only one include, then stick with this but make clear that it is only applicable in that case.